tests/glsl-intrinsic/atomic/atomicStorageBuffer.slang

```diff
@@ -0,0 +1,318 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl -DTARGET_GLSL
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv -emit-spirv-directly -DTARGET_SPIRV
+
+// currently tests do not handle atomic_uint well
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly 
+#version 430
+
+
+//#define TEST_when_shader_atomic_float2_is_available
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
+buffer MyBlockName
+{
+    uint data[1];
+} outputBuffer;
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=int32Buffer
+buffer MyBlockName1
+{
+    int data[1];
+} int32Buffer;
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=8):out,name=int64Buffer
+buffer MyBlockName2
+{
+    uint64_t data[1];
+} int64Buffer;
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=uint32Buffer
+buffer MyBlockName3
+{
+    uint data[1];
+} uint32Buffer;
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=8):out,name=uint64Buffer
+buffer MyBlockName4
+{
+    uint64_t data[1];
+} uint64Buffer;
+
+//TEST_INPUT:ubuffer(data=[0.0f], stride=2):out,name=float16Buffer
+buffer MyBlockName5
+{
+    half data[1];
+} float16Buffer;
+
+//TEST_INPUT:ubuffer(data=[0.0f], stride=2):out,name=float32Buffer
+buffer MyBlockName6
+{
+    float data[1];
+} float32Buffer;
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=8):out,name=float64Buffer
+buffer MyBlockName7
+{
+    double data[1];
+} float64Buffer;
+
+bool testAtomicInt32()
+{
+    return true
+        && (int32Buffer.data[0] = 5) == 5
+        && atomicAdd(int32Buffer.data[0], 1) == 5
+        && int32Buffer.data[0] == 6
+
+        && (int32Buffer.data[0] = 5) == 5
```

Comment:
we can't chain values because that would imply that every result must be the same (else inputs will be inconsistent when removing individual cases)

if every result is the same, we have no idea if the code is working, or doing nothing, since the tests are testing a scenario when both are possible

Reply:
What I meant was more like this:
```
        && (int32Buffer.data[0] = 5) == 5
        && atomicAdd(int32Buffer.data[0], 1) == 5
        && int32Buffer.data[0] == 6
        && atomicMin(int32Buffer.data[0], 1) == 6
        && int32Buffer.data[0] == 1
        && atomicMax(int32Buffer.data[0], 5) == 5
        && int32Buffer.data[0] == 5
         ... and so on ....
```

Reply:
to feed last operations result into the next we need to assume the input->output will be the same always. 

the following will work predictably regardless of which block is commented out
```
        && (int32Buffer.data[0] = 5) == 5
        && atomicAnd(int32Buffer.data[0], 1) == 5
        && int32Buffer.data[0] == 1

        && (int32Buffer.data[0] = 5) == 5
        && atomicOr(int32Buffer.data[0], 2) == 5
        && int32Buffer.data[0] == 7

        && (int32Buffer.data[0] = 5) == 5
        && atomicXor(int32Buffer.data[0], 3) == 5
        && int32Buffer.data[0] == 6
```

but if we remove the setter we will have something like this which would work:
```
        && (int32Buffer.data[0] = 5) == 5
        && atomicAnd(int32Buffer.data[0], 1) == 5
        && int32Buffer.data[0] == 1

        && atomicOr(int32Buffer.data[0], 2) == 1
        && int32Buffer.data[0] == 2

        && atomicXor(int32Buffer.data[0], 3) == 2
        && int32Buffer.data[0] == 1
```

but if we comment out `Or` to test And and Xor we will get a test that is checking the wrong values:
```
        && (int32Buffer.data[0] = 5) == 5
        && atomicAnd(int32Buffer.data[0], 1) == 5
        && int32Buffer.data[0] == 1

        && atomicXor(int32Buffer.data[0], 3) == 2
        && int32Buffer.data[0] == 1
```

Reply:
👍